### DISCUSSION TITLE
Can I use Booster in combination with Serverless Framework (serverless.yml file)?

### DISCUSSION THREAD

**Discussion by fernando1989mg** [boostercloud/booster#613] [Q&A]
i want to use booster to migrate a service... but actually i have important configuration in my serverless.yml, so, how i can use Booster and my serverless.yml together?

**Comment by javiertoledo** [comment]
Hello @fernando1989mg, the main difference between Booster and Serverless Framework is that Booster focuses on the application code and generates the infra automatically, while Serverless Framework focuses on the infrastructure components, leaving the code responsibilities to the developer. In a sense, they solve different problems, so it could make sense to use both.

Booster also deploys your application on top of Lambda, DynamoDB and API Gateway, it's just that it does it in a more implicit way. So, at the end of the day, it's all AWS, and nothing stops you from deploying one part of your system using Booster and other using the Serverless framework. You'll just end up with two (or more dependeing on your config) separate cloudformation stacks, the one deployed by Booster and the one deployed by SLS framework and you can make them interoperate as with any other serverless application.

Depending on your use case, you have different options: 
* Probably the easiest way to make a Booster app play nicely with a Serverless app would be to keep them fully isolated and use the APIs provided by each service from a shared client application as two separate microservices, making any integration at the client level (you can easily share a single JWT token provider for both applications to make this easier). 
* The next "easiest" way could be using a SQS queue (or other intermediate service) and use the AWS SDK from Booster handlers to push the data you want to send to the Serverless app. You can also subscribe a serverless function to the Booster app's events table using DynamoDB streams (you can find it in the AWS console after deployment). The best way to "send messages" the other way would be injecting an event in the Booster's DynamoDB events table and creating an event handler. Booster triggers the event handling pipeline on every write to the events table.
* If you need more advanced integrations in the Booster side (like using a different database, or react to other AWS events), the framework is designed in a highly modular way, so you can fork or replace [the current AWS implementation](https://github.com/boostercloud/booster/tree/main/packages/framework-provider-aws-infrastructure) with your own, or [create a rocket](https://github.com/boostercloud/booster/tree/main/docs#extending-booster-with-rockets) (basically a plugin) that extends the application infrastructure using the AWS CDK.

I'm not sure if these ideas help, we've been focusing on using Booster with the AWS CDK, but that's only one way of dealing with infrastructure in AWS, and the framework is designed to support other implementations. It could be totally possible to create a new `ProviderLibrary` implementation that uses Serverless Framework under the hoods to create the infrastructure. It's a bit of extra work, but is not that complex; a provider library is just a package that implements [the `ProviderLibrary` interface](https://github.com/boostercloud/booster/blob/main/packages/framework-types/src/provider.ts). 

There's a prototype implementation that @rdiaz82 created to support [DAPR](https://dapr.io/) integrations in [the kubernetes provider](https://github.com/boostercloud/booster/tree/main/packages/framework-provider-kubernetes) that could be used as a reference to do something this. We don't have plans to build a Serverless integration yet, but it's certainly of our interest to support new ways to deploy Booster apps, so if someone wants to try, they'll have our full support!
